app/controllers/actions/IdentifierAction.scala

```diff
@@ -55,59 +59,62 @@ class AuthenticatedIdentifierAction @Inject()(
 
   override protected def refine[A](request: Request[A]): Future[Either[Result, IdentifierRequest[A]]] = {
 
-    implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromHeadersAndSession(request.headers, None)
+    implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromHeadersAndSession(request.headers, Some(request.session))
+
+    authorised(User).retrieve(
+      Retrievals.allEnrolments and
+        Retrievals.credentials and
+        Retrievals.affinityGroup
+    ) {
+      case enrolments ~ Some(credentials) ~ Some(affinityGroup) =>
+        Logger.debug("DefaultAuthAction:Refine - Enrolments:" + enrolments)
 
-    authorised(User).retrieve(Retrievals.allEnrolments and Retrievals.credentials and Retrievals.affinityGroup) {
-      case enrolments ~ Some(credentials) ~ affinityGroup =>
-        enrolmentMessage("AuthenticatedIdentifierAction:Refine - Enrolments:", Some(enrolments))
-        Future.successful(Right(IdentifierRequest(request, credentials.providerId, affinityGroup)))
+        Future.successful(
+          Right(
+            IdentifierRequest(
+              request,
+              amlsRefNo(enrolments),
+              credentials.providerId,
+              affinityGroup
+            )
+          )
+        )
       case _ =>
-        enrolmentMessage("AuthenticatedIdentifierAction:Refine - Non match (enrolments ~ Some(credentials) ~ affinityGroup)", None)
+        Logger.debug("DefaultAuthAction:Refine - Non match (enrolments ~ Some(credentials) ~ Some(affinityGroup))")
         Future.successful(Left(Redirect(Call("GET", config.loginUrl))))
     }.recover[Either[Result, IdentifierRequest[A]]] {
       case nas: NoActiveSession =>
-        exceptionLogger(nas)
+        Logger.debug("DefaultAuthAction:Refine - NoActiveSession:" + nas)
         Left(Redirect(Call("GET", config.loginUrl)))
       case ie: InsufficientEnrolments =>
-        exceptionLogger(ie)
+        Logger.debug("DefaultAuthAction:Refine - InsufficientEnrolments:" + ie)
```

Comment:
before logging was done with use of enrolmentMessage and exceptionLogger methods, the reason behind that was to exclude debug statements from code coverage analysis in this file.